include/ccmath/math/nearest/nearbyint.hpp

```diff
@@ -8,7 +8,64 @@
 
 #pragma once
 
+#include <ccmath/math/compare/isinf.hpp>
+#include <ccmath/math/compare/isnan.hpp>
+#include <ccmath/math/compare/signbit.hpp>
+#include <ccmath/math/nearest/trunc.hpp>
+#include <type_traits>
+
 namespace ccm
 {
+	/**
+	 * @brief The nearest integer value to num, according to the rounding mode FE_TONEAREST, is returned.
+	 * @tparam T The type of the number.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the rounded floating point value. Otherwise, returns the input floating point value unmodified.
+	 */
+	template <class T, std::enable_if_t<std::is_floating_point_v<T>, bool> = true>
+	constexpr T nearbyint(T num) noexcept
+	{
+		// If num is ±∞ or ±0, num is returned, unmodified.
+		if (ccm::isnan(num) || ccm::isinf(num) || num == T(0)) { return num; }
+
+		T int_part	= ccm::trunc(num);
+		T frac_part = num - int_part;
+		if (frac_part == T(0.0)) { return int_part; }
+		if (frac_part > T(0.5) || (frac_part == T(0.5) && static_cast<long long>(int_part) % 2 != 0)) { return int_part + T(1); }
```

Comment:
Casting to long long here introduces some risks due to the implementation-defined behavior across different platforms. While long long is guaranteed to be at least 64 bits, the cast may lead to precision loss or unexpected behavior, especially when dealing with long double. This issue might not manifest on all platforms—on some, the cast may work as expected—but it’s safer to avoid relying on this behavior. Instead, we should consider using our helper class `ccmath/tree/main/include/ccmath/internal/support/fp/fp_bits.hpp` to directly manipulate the floating-point bits. This approach will ensure more consistent and reliable handling, particularly for long double, and mitigate the risks associated with platform-specific integer representations. The class also handles a lot of the tricky bits in bit manipulation also. ^^

Reply:
Thanks! I will look into replacing this.